File: backend/app/schemas/prompt.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, Field


class PromptVersionOut(BaseModel):
    id: str
    prompt_id: str
    version_number: int
    content: str
    variables: Optional[list[str]] = None
    is_current: bool
    created_by: Optional[str]
    created_at: Optional[datetime]

    model_config = {"from_attributes": True}


class PromptOut(BaseModel):
    id: str
    tenant_id: str
    name: str
    description: Optional[str]
    category: Optional[str]
    tags: Optional[list[str]]
    status: str
    created_by: Optional[str]
    created_at: Optional[datetime]
    updated_at: Optional[datetime]
    current_version: Optional[PromptVersionOut] = None

    model_config = {"from_attributes": True}


class PromptCreate(BaseModel):
    name: str = Field(..., max_length=255)
    description: Optional[str] = None
    category: Optional[str] = Field(None, max_length=100)
    tags: Optional[list[str]] = None
    content: str = Field(..., description="Initial prompt content, may contain {{variable}} placeholders")


class PromptUpdate(BaseModel):
    name: Optional[str] = Field(None, max_length=255)
    description: Optional[str] = None
    category: Optional[str] = Field(None, max_length=100)
    tags: Optional[list[str]] = None
    status: Optional[str] = Field(None, pattern="^(draft|published|archived)$")


class PromptVersionCreate(BaseModel):
    content: str = Field(..., description="New version content")


class PromptTestRequest(BaseModel):
    version_id: Optional[str] = None
    variables: dict[str, str] = Field(default_factory=dict)
    model_id: str


class PromptTestResult(BaseModel):
    rendered_content: str
    result_content: str
    prompt_tokens: int
    completion_tokens: int
    latency_ms: int

File: frontend/src/types/prompt.ts
export type PromptStatus = 'draft' | 'published' | 'archived'

export interface PromptVersion {
  id: string
  prompt_id: string
  version_number: number
  content: string
  variables: string[] | null
  is_current: boolean
  created_by: number | null
  created_at: string | null
}

export interface Prompt {
  id: string
  tenant_id: string
  name: string
  description: string | null
  category: string | null
  tags: string[] | null
  status: PromptStatus
  created_by: number | null
  created_at: string | null
  updated_at: string | null
  current_version: PromptVersion | null
}

export interface PromptCreateRequest {
  name: string
  description?: string
  category?: string
  tags?: string[]
  content: string
}

export interface PromptUpdateRequest {
  name?: string
  description?: string
  category?: string
  tags?: string[]
  status?: PromptStatus
}

export interface PromptVersionCreateRequest {
  content: string
}

export interface PromptTestRequest {
  version_id?: string
  variables: Record<string, string>
  model_id: string
}

export interface PromptTestResult {
  rendered_content: string
  result_content: string
  prompt_tokens: number
  completion_tokens: number
  latency_ms: number
}
